src/lib/db/project-store.ts

```diff
@@ -44,13 +54,62 @@ class ProjectStore implements IProjectStore {
 
     async exists(id: string): Promise<boolean> {
         const result = await this.db.raw(
-            `SELECT EXISTS (SELECT 1 FROM ${TABLE} WHERE id = ?) AS present`,
+            `SELECT EXISTS(SELECT 1 FROM ${TABLE} WHERE id = ?) AS present`,
             [id],
         );
         const { present } = result.rows[0];
         return present;
     }
 
+    async getProjectsWithCounts(
+        query?: IProjectQuery,
+    ): Promise<IProjectWithCount[]> {
+        const projectTimer = this.timer('getProjectsWithCount');
+        let projectsQuery = this.db(TABLE)
+            .select(
+                this.db.raw(
+                    'projects.id, projects.name, projects.description, projects.health, projects.updated_at, count(features.name) AS number_of_features',
+                ),
+            )
+            .leftJoin('features', 'features.project', 'projects.id')
+            .groupBy('projects.id');
+        if (query) {
+            projectsQuery = projectsQuery.where(query);
+        }
+        const projectAndFeatureCount = await projectsQuery;
+
+        // @ts-ignore
+        const projectsWithFeatureCount = projectAndFeatureCount.map(
+            this.mapProjectWithCountRow,
+        );
+        projectTimer();
+        const memberTimer = this.timer('getMemberCount');
+        const memberCount = await this.db.raw(
+            `SELECT count(role_id) as member_count, project FROM role_user GROUP BY project`,
+        );
+        memberTimer();
+        const memberMap = memberCount.rows.reduce((a, r) => {
+            a[r.project] = parseInt(r.member_count, 10);
```

Comment:
Do we need the radix here? The two cases I know off the top of my head are on old JS interpreters with being too eager to interpret this as hex and passing parseInt to be invoked by a map/reduce/filter. I don't think we have either case here.

But do we want it for explicit's sake?

Reply:
Usually knex have handled this when typing the rows already, so it shouldn't be necessary, but I had tests failiing, this solved it, and I like to be explicit with radixes when using parseInt anyway, just so nobody thinks I've got a funky idea about base 8 :)

Reply:
Let me have a go at writing it in knex syntax instead to see if that is better

Reply:
Ah, Note that in Postgres, count returns a bigint type which will be a String and not a Number, that explains why it became a string. Fair enough, then we'll still need this parsing. We'll have performance problems way before we hit javascript accuracy issues due to reducing a bigint to int.

Reply:
Maybe better to just use the `+` in front?

Reply:
Thanks @ivarconr that seems better :)